functions/src/resolve-market.ts

```diff
@@ -15,156 +15,146 @@ import {
-          outcome,
-          resolutions ?? {},
-          contract,
-          bets,
-          liquidities,
-          resolutionProbability
-        )
-
-      await contractDoc.update(
-        removeUndefinedProps({
-          isResolved: true,
-          resolution: outcome,
-          resolutionValue: value,
-          resolutionTime,
-          closeTime: newCloseTime,
-          resolutionProbability,
-          resolutions,
-          collectedFees,
-        })
-      )
+  const { payouts, creatorPayout, liquidityPayouts, collectedFees } =
+    getPayouts(
+      outcome,
+      Object.fromEntries(resolutions || []),
+      contract,
+      bets,
+      liquidities,
+      resolutionProbability
+    )
 
-      console.log('contract ', contractId, 'resolved to:', outcome)
+  await contractDoc.update(
+    removeUndefinedProps({
+      isResolved: true,
+      resolution: outcome,
+      resolutionValue: value,
+      resolutionTime,
+      closeTime: newCloseTime,
+      resolutionProbability,
+      resolutions,
+      collectedFees,
+    })
+  )
 
-      const openBets = bets.filter((b) => !b.isSold && !b.sale)
-      const loanPayouts = getLoanPayouts(openBets)
+  console.log('contract ', contractId, 'resolved to:', outcome)
 
-      if (!isProd())
-        console.log(
-          'payouts:',
-          payouts,
-          'creator payout:',
-          creatorPayout,
-          'liquidity payout:'
-        )
+  const openBets = bets.filter((b) => !b.isSold && !b.sale)
+  const loanPayouts = getLoanPayouts(openBets)
 
-      if (creatorPayout)
-        await processPayouts(
-          [{ userId: creatorId, payout: creatorPayout }],
-          true
-        )
+  if (!isProd())
+    console.log(
+      'payouts:',
+      payouts,
+      'creator payout:',
+      creatorPayout,
+      'liquidity payout:'
+    )
 
-      await processPayouts(liquidityPayouts, true)
+  if (creatorPayout)
+    await processPayouts([{ userId: creatorId, payout: creatorPayout }], true)
 
-      const result = await processPayouts([...payouts, ...loanPayouts])
+  await processPayouts(liquidityPayouts, true)
 
-      const userPayoutsWithoutLoans = groupPayoutsByUser(payouts)
+  await processPayouts([...payouts, ...loanPayouts])
 
-      await sendResolutionEmails(
-        openBets,
-        userPayoutsWithoutLoans,
-        creator,
-        creatorPayout,
-        contract,
-        outcome,
-        resolutionProbability,
-        resolutions
-      )
+  const userPayoutsWithoutLoans = groupPayoutsByUser(payouts)
 
-      return result
-    }
+  await sendResolutionEmails(
+    openBets,
+    userPayoutsWithoutLoans,
+    creator,
+    creatorPayout,
+    contract,
+    outcome,
+    resolutionProbability,
+    Object.fromEntries(resolutions || [])
   )
 
+  const updatedContractSnap = await contractDoc.get()
+  if (!updatedContractSnap)
+    throw new APIError(500, 'Updated contract does not exist')
+  return updatedContractSnap.data() as Contract
```

Comment:
Kind of ridiculous to fetch it again from the DB when we just put this data. I would prefer to either return the data as we have it in memory, or just return the ID/slug and let the user look it up if they want the new data.